src/ServicePulse.Host/vue/src/components/monitoring/D3Graph.vue

```diff
@@ -0,0 +1,141 @@
+<script setup>
+import { ref, onMounted } from "vue";
+import * as d3 from "d3";
+/*import * as d3 from "https://cdn.jsdelivr.net/npm/d3@7/+esm";*/
+import { useFormatTime, useFormatLargeNumber } from "../../composables/formatter.js";
+const props = defineProps({
+  plotdata: Object,
+  minimumyaxis: Number,
+  avglabelcolor: String,
+  isdurationgraph: Boolean,
+  metricsuffix: String,
+  csclass: String,
+  endpointname: String,
+  colname: String,
+});
+const averageDecimalsDefault = 2;
+const avgLabelColorDefault = "#2700CB";
+const avgLabelSuffixDefault = "";
+const root = ref(null);
+
+function displayGraphValues() {
+  var svg = root.value.getElementsByTagName("svg")[0];
+  var width = svg.clientWidth;
+  var height = svg.clientHeight;
+  //HINT: This is workaround for Firefox
+  if (width === 0) {
+    var box = svg.getBoundingClientRect();
+
+    width = box.right - box.left;
+    height = box.bottom - box.top;
+  }
+
+  var verticalMargin = 6;
+  var horizontalMargin = 2;
+  var avgDecimals = avgDecimals || averageDecimalsDefault;
+  var avgLabelColor = props.avglabelcolor || avgLabelColorDefault;
+  var metricSuffix = props.metricsuffix || avgLabelSuffixDefault;
+  var plotData = props.plotdata || { points: [], average: 0 };
+  var points = plotData.points;
+  if (points.length === 0) {
+    points = new Array(10).fill(0);
+  }
+  var average = plotData.average || 0;
+  var minimumYaxis = !isNaN(props.minimumyaxis) ? Number(props.minimumyaxis) : 10;
+  var max = points && points.length ? Math.max(average * 1.5, d3.max(points), minimumYaxis) : 1;
+  var numberOfPoints = points && points.length ? points.length : 2;
+
+  var scaleY = d3
+    .scaleLinear()
+    .domain([0, max])
+    .range([height - verticalMargin, verticalMargin]);
+
+  var scaleX = d3
+    .scaleLinear()
+    .domain([0, numberOfPoints - 1])
+    .range([horizontalMargin, width - horizontalMargin]);
+
+    var area = d3
+        .area()
+        .x(function (d, i) {
+            console.log(d); //dummy call to get past eslint error
```

Comment:
Could we try something to avoid leaving the console.log? Like using deconstructing for assigning the non-used parameter to a lowercase dash _ and then configuring lint options to allow for that pattern? https://eslint.org/docs/latest/rules/no-unused-vars#varsignorepattern https://eslint.org/docs/latest/rules/no-unused-vars#argsignorepattern

Reply:
Agreed. I will work on it in a separate PR

Reply:
Sounds good. Approving now. Let's add a task so we don't forget to configure ling and remove the console.log